yo1k/tic_tac_toe/core/game.py

```diff
@@ -0,0 +1,296 @@
+        `round` keeping the same `Player` index in the `State.players`.
+        """
+        return (state.step + state.round) % 2
+
+    @staticmethod
+    def __occupy(state: State, cell: Cell):
+        turn = Logic.__turn(state)
+        mark = state.players[turn].mark()
+        if state.board._get(cell) is None:
+            state.board._set(cell, mark)
+        else:
+            pass
+        if Logic._win_condition(state.board, cell):
+            Logic.__win(state)
+        elif Logic._draw_condition(state.step):
+            Logic.__draw(state)
+        else:
+            state.step += 1
+
+    @staticmethod
+    def _win_condition(board: Board, cell: Cell) -> bool:
+        win_condition = False
+        horizontal_match_number = 0
+        vertical_match_number = 0
+        first_diagonal_match_number = 0
+        second_diagonal_match_number = 0
+        x = cell.x
+        y = cell.y
+        mark = board._get(Cell(x, y))
+        for i in range(Board.size()):
+            if board._get(Cell(i, y)) == mark:
+                horizontal_match_number += 1
+            if board._get(Cell(x, i)) == mark:
+                vertical_match_number += 1
+            if board._get((Cell(i, i))) == mark:
+                first_diagonal_match_number += 1
+            if board._get(Cell(i, Board.size() - 1 - i)) == mark:
+                second_diagonal_match_number += 1
+        if horizontal_match_number == Board.size() or \
+           vertical_match_number == Board.size() or \
+           first_diagonal_match_number == Board.size() or \
+           second_diagonal_match_number == Board.size():
+            win_condition = True
+        return win_condition
+
+    @staticmethod
+    def _draw_condition(step: int) -> bool:
+        draw_condition = False
+        if step == Board.size() ** 2 - 1:
+            draw_condition = True
+        return draw_condition
+
+    @staticmethod
+    def __surrender(state: State):
+        index_other_player = (Logic.__turn(state) + 1) % 2
+        state.players[index_other_player].wins += 1
+        state.phase = Phase.OUTROUND
+
+    @staticmethod
+    def __win(state: State):
+        state.players[Logic.__turn(state)].wins += 1
+        state.phase = Phase.OUTROUND
+
+    @staticmethod
+    def __draw(state: State):
+        state.phase = Phase.OUTROUND
+
+    @staticmethod
+    def __start(state: State):
+        if state.phase is Phase.OUTROUND:
+            state.round += 1
+            state.step = 0
+            state.board._clean()
+        elif state.phase is Phase.ONSET:
+            pass
+        else:
+            assert False
+        state.phase = Phase.INROUND
+
+    def __str__(self):
+        return f"{type(self)}(" \
+               f"action_queues={self.__action_queues})"
+
+
+class World:
+    def __init__(self, state: State, logic: Logic):
+        self.__state = state
+        self.__logic = logic
+
+    def run(self):  # SKTODO make method
+        pass
```

Comment:
We agreed that `TODO` tags should not appear in a PR.

`World` simply combines ` State` and `Logic`. It needs a public method `advance` that accepts nothing and calls `logic.advance(state)`.

Reply:
Done.